docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -346,6 +346,42 @@ and the lib version (in ``"<api version>[.<patch version>]"`` string format).
 
 .. literalinclude:: charmcraft-sample-charm.yaml
     :start-at: charm-libs:
+    :end-before: charm-user:
+
+
+.. _charmcraft-yaml-key-charm-user:
+
+``charm-user``
+--------------
+
+.. important::
+
+    ``charm-user`` was added in Juju 3.6.0. Currently only supported on Kubernetes
+    charms and has no effect on machine charms.
+
+**Status:** Optional. Recommended for Kubernetes charms.
+
+**Purpose:** The ``charm-user`` key  allows charm authors to specify that their charm
+hook code does not need to be run as root. This key, in addition to ``uid`` + ``uid``
+fields in ``containers``, allows charms to be run rootless. The value of ``root``
+ensures the charm runs as root. Both ``sudoer`` and ``non-root`` will run as a user
+other than root. In the case of the value ``sudoer``, the charm will be run as a user
+with access to sudo to elevate its privileges.
```

Comment:
Here's an attempt at tidying up some of the language. It contains a mix of style guide conventions and semantic changes.

I assumed this...

> ``uid`` + ``uid``

... is a typo, hence the change.

```suggestion
**Purpose:** The ``charm-user`` key allows charm authors to specify that their charm
hook code doesn't need to run as root. This key, in addition to the ``containers.uid``
and ``containers.gid`` keys, allows the charm to be run rootless. If set to ``root``,
the charm runs as root. If set to ``sudoer`` or ``non-root``, the charm runs as a user
other than root. If the value is ``sudoer``, the charm runs as a superuser, giving it
elevated privileges.
```

Reply:
This is all copied from the `metadata.yaml` definition. Do you want it changed there as well?

"does not need to be run as root" does not have the same meaning as "doesn't need root access", so that would be a bad change. "runs as a superuser" also does not have the same meaning as "run as a user with access to sudo", so that would also be a bad change.